modules/kodi.py

```diff
@@ -885,4 +886,9 @@ def url(self, path='', auth=False):
     def auth(self):
         """ Generate a base64 HTTP auth string based on settings """
         if self.current.username and self.current.password:
-            return base64.encodestring('%s:%s' % (self.current.username, self.current.password)).strip('\n')
+            dummy_param = 'bla'
+            key_url_encoded = urlencode({dummy_param: self.current.username})[len(dummy_param) + 1:]
+            secret_url_encoded = urlencode({dummy_param: self.current.password})[len(dummy_param) + 1:]
+            credentials = '{}:{}'.format(key_url_encoded, secret_url_encoded)
+            bytes_base64_encoded_credentials = base64.encodebytes(credentials.encode('utf-8'))
+            return bytes_base64_encoded_credentials.decode('utf-8').replace('\n', '')
```

Comment:
Here the cleaner solution 🙂:
```suggestion
            credentials = f'{quote_plus(self.current.username)}:{quote_plus(self.current.password)}'
            return base64.encodebytes(credentials.encode()).decode('utf-8').replace('\n', '')
```